pages/dashboard/DisplayInvitations.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useRouter } from 'next/router';
+
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import TextButton from '../../components/UI/TextButton';
+import AcceptIcon from '../../assets/icons/accept.svg';
+import CloseIcon from '../../assets/icons/close.svg';
+import LoadingSpinnerInline from '../../components/UI/LoadingSpinnerInline';
+
+/**
+ * COMPONENT 'DisplayInvitations'
+ * Displays an invitation for a matrix room/space within a <ServiceTable> component.
+ *
+ * @param {Object} invite — object of the room the user was invited to
+ * @param {String} service — name of the service (i.e. the object key names of the 'authProvider' object in the config)
+ * @param {String} name — name of the Application (i.e. the 'path' variable in the config)
+ *
+ * @callback acceptMatrixInvite - callback function to accept invitations to a matrix room or space
+ * @param e - event triggered by the button
+ * @param roomId - matrix roomId
+ * @param service - name of the service (parsed to the function)
+ *
+ * @callback declineMatrixInvite - callback function to decline invitations to a matrix room or space
+ * @param e - event triggered by the button
+ * @param roomId - matrix roomId
+ *
+ * @returns {React.ReactComponent}
+*/
+
+const DisplayInvitations = ({ invite, name, service, acceptMatrixInvite, declineMatrixInvite }) => {
+    const { t } = useTranslation('dashboard');
+    const [isAcceptingInvite, setIsAcceptingInvite] = useState(false);
+    const [isDecliningInvite, setisDecliningInvite] = useState(false);
+    const router = useRouter();
+
+    const handleDecline = async (e, roomId) => {
+        setisDecliningInvite(true);
+        await declineMatrixInvite(e, roomId);
+        setisDecliningInvite(false);
+    };
+
+    const handleAccept = async (e, roomId) => {
+        setIsAcceptingInvite(true);
+        await acceptMatrixInvite(e, roomId, service);
+        setIsAcceptingInvite(false);
+        if (confirm('You’ve successfully accepted the invitation!\n\nWould you like to be redirected to the newly accepted ' + name + ' item?')) {
+            router.push(`${name}/${roomId}`);
```

Comment:
Instead of _assuming_ that this path pattern always works, can we make the corresponding consumer define it? Something I can think of, for example, would be to return it in the `acceptMatrixInvite` function like this:
```js
const forwardingUrl = await acceptMatrixInvite(e, roomId, service);

...

if (...) { router.push(forwardingUrl); }
```